hushline/premium.py

```diff
@@ -0,0 +1,609 @@
+
+            current_app.logger.info(
+                f"Processing event {stripe_event.event_type} ({stripe_event.event_id})"
+            )
+            try:
+                # subscription events
+                if event.type.startswith("customer.subscription."):
+                    subscription: stripe.Subscription = stripe.Subscription.construct_from(
+                        event.data.object, current_app.config["STRIPE_SECRET_KEY"]
+                    )
+                    if event.type == "customer.subscription.created":
+                        handle_subscription_created(subscription)
+                    elif event.type == "customer.subscription.updated":
+                        handle_subscription_updated(subscription)
+                    elif event.type == "customer.subscription.deleted":
+                        handle_subscription_deleted(subscription)
+                # invoice events
+                elif event.type.startswith("invoice."):
+                    invoice: stripe.Invoice = stripe.Invoice.construct_from(
+                        event.data.object, current_app.config["STRIPE_SECRET_KEY"]
+                    )
+                    if event.type == "invoice.created":
+                        handle_invoice_created(invoice)
+                    elif event.type == "invoice.updated":
+                        handle_invoice_updated(invoice)
+
+            except Exception as e:
+                current_app.logger.error(
+                    f"Error processing event {stripe_event.event_type} ({stripe_event.event_id}): {e}\n{stripe_event.event_data}"  # noqa: E501
+                )
+                stripe_event.status = StripeEventStatusEnum.ERROR
+                stripe_event.error_message = str(e)
+                db.session.add(stripe_event)
+                db.session.commit()
+                continue
+
+            stripe_event.status = StripeEventStatusEnum.FINISHED
+            db.session.add(stripe_event)
+            db.session.commit()
+
+
+def create_blueprint(app: Flask) -> Blueprint:
+    # Now define the blueprint
+    bp = Blueprint("premium", __file__, url_prefix="/premium")
+
+    @bp.route("/", methods=["GET"])
+    @authentication_required
+    def index() -> Response | str:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        # Check if we have an incomplete subscription
+        stripe_subscription = get_subscription(user)
+        if stripe_subscription and stripe_subscription["status"] == "incomplete":
+            flash("⚠️ Your subscription is incomplete. Please try again.", "warning")
+
+        # Load the user's invoices
+        invoices = db.session.scalars(
+            db.select(StripeInvoice)
+            .filter_by(user_id=user.id)
+            .filter_by(status=StripeInvoiceStatusEnum.PAID)
+            .order_by(desc(StripeInvoice.created_at))
```

Comment:
Nit. Can be `.order_by(StripeInvoice.created_at.desc())`

Reply:
Fixed in 01603c0d5b5b23eec26e23dae75c84dfc1fd839b